docs/proposals/petset.md

```diff
@@ -0,0 +1,533 @@
+the cluster if the PV for identities 4 and 5 are deleted (may not be recoverable). For the initial proposal,
+leaving the supporting resources is the safest path (safety first) with a potential future policy applied
+to the PetSet for how to manage supporting resources (DeleteImmediately, GarbageCollect, Preserve).
+
+OPEN: Should information about the state of the pods in the PetSet be reflected in PetSet status? Summarizing
+  which pods match the current PetSet state would be extremely valuable.
+
+
+### Parameterizing pod templates and supporting resources
+
+Since each pod needs a unique and distinct identity, and the pod needs to know its own identity, the
+PetSet must allow a pod template to be parameterized by the identity assigned to the pod. The pods that
+are created should be easily identified by their cluster membership.
+
+
+```
+type PetSetSpec struct {
+    // Number of instances in the set
+    Replicas *int
+
+    // The template to use for the pod in the set
+    Template *PodTemplateSpec
+}
+
+// Proposal 1a: Use the name of the pod as the identity, use downward API for getting identity
+// Downside: all fields that want to leverage the identity need equivalent valueFrom semantics
+{
+    "spec": {
+        "template": {
+            "metadata": {"generateName": "pod-name"}, // assume identity is automatically added to pod name
+            "spec": {
+                "containers": [{
+                    "env": [{
+                        "name": "IDENTITY",
+                        "valueFrom": {"fieldRef": {"fieldPath": "metadata.name"}
+                        }
+                    }]
+                }]
+            }
+        }
+    }
+}
+
+// Proposal 1b: Add an annotation to each pod when the controller creates it
+// Downside: all fields that want to leverage the identity need equivalent valueFrom semantics
+{
+    "spec": {
+        "template": {
+            "spec": {
+                "containers": [{
+                    "env": [{
+                        "name": "IDENTITY",
+                        "valueFrom": {"fieldRef": {"fieldPath": "metadata.annotations[alpha.kubernetes.io/identity]"}}
+                    }]
+                }]
+            }
+        }
+    }
+}
+
+// Proposal 1c: Add a new field to pod spec that is write-once/read-many - identity
+// Downside: all fields that want to leverage the identity need equivalent valueFrom semantics
+{
+    "spec": {
+        "template": {
+            "spec": {
+                "identity": "", // set on creation
+                "containers": [{
+                    "env": [{
+                        "name": "IDENTITY",
+                        "valueFrom": {"fieldRef": {"fieldPath": "spec.identity"}}
+                    }]
+                }]
+            }
+        }
+    }
+}
+
+// Proposal 1d: support template transformation of a field path
+// Downside: possibly too flexible, can conflict with existing env resolution semantics, bypasses downward API
```

Comment:
I wasn't arguing that we shouldn't be able to, I was arguing that it feels like it should be a case of "every field an be templated" or maybe "every string field".  But some fields are already expanded for env, so we at least need to define the expansion to be compatible and make sure the intersection is sensible.

As long as we kept it SUPER restricted - strings only - it could work.  It has the advantage of not needing a `valueFrom` for every field.  It is a precedent, though, so tread cautiously.